web/src/state_data.ts

```diff
@@ -153,6 +153,24 @@ export const muted_user_schema = z.object({
     timestamp: z.number(),
 });
 
+const unread_stream_info_schema = z.object({
+    stream_id: z.number(),
+    topic: z.string(),
+    unread_message_ids: z.array(z.number()),
+});
+
+export const unread_direct_message_info_schema = z.object({
+    other_user_id: z.number(),
+    // Deprecated and misleading synonym for other_user_id
+    sender_id: z.number(),
```

Comment:
I think we could move the web app to just ignoring the deprecated `sender_id` field, after making sure downstream code isn't relying on it existing. I'm pretty sure that's the intent for clients that know to check `other_user_id`.